TenantFile/Api/Controllers/SmsController.cs

```diff
@@ -68,25 +78,67 @@ await document.Reference.UpdateAsync("Messages", FieldValue.ArrayUnion(new Dicti
 
             var filenames = await SaveMedia(numMedia);
 
+
+            var phone = _context.Phones.FirstOrDefault(x => x.PhoneNumber == request.From);
+
+            if (phone == null)
+            {
+                phone = new Phone
+                {
+                    PhoneNumber = request.From,
+                    Images = new List<Models.Image>()
+                };
+                _context.Phones.Add(phone);
+            }
+
+            foreach (var (image, thumbnail) in filenames)
+            {
+                if (phone.Images == null)
+                {
+                    phone.Images = new List<Models.Image>();
+                }
+                phone.Images.Add(new Models.Image
+                {
+                    Name = image,
+                    ThumbnailName = thumbnail
+                });
+            }
+
+            _context.SaveChanges();
+
             var response = new MessagingResponse();
             var messageBody = numMedia == 0 ? "Send us an image!" :
                 $"Thanks for sending us {numMedia} file(s)!";
             response.Message(messageBody);
             return TwiML(response);
         }
 
-        private async Task<IEnumerable<string>> SaveMedia(int numMedia)
+        private async Task<IEnumerable<(string image, string thumbnail)>> SaveMedia(int numMedia)
         {
-            var filenames = new List<string>();
+            var filenames = new List<(string, string)>();
             for (var i = 0; i < numMedia; i++)
             {
                 var mediaUrl = Request.Form[$"MediaUrl{i}"];
                 _logger.LogInformation(mediaUrl);
                 var contentType = Request.Form[$"MediaContentType{i}"];
 
                 var filePath = Path.Combine("images", GetMediaFileName(mediaUrl, contentType));
-                filenames.Add(filePath);
+
                 await _storageClient.UploadToStorageAsync(mediaUrl, filePath, contentType);
+
+                var httpClient = new HttpClient();
+                var response = await httpClient.GetAsync(mediaUrl);
+                var inputStream = await response.Content.ReadAsStreamAsync();
+                using var image = SixLabors.ImageSharp.Image.Load(inputStream);
+                image.Mutate(x => x
+                     .Resize(0, 100));
```

Comment:
We should resize then cut a square or some other shape out of the middle so all the thumbnails are uniform?

Reply:
Yeah, I was pretty lazy with the thumbnail algorithm. We should probably find the shortest dimension, resize to 100 (or something constant), then cut to a square like you said.